Source/Scene/GroundPolygon.js

```diff
@@ -0,0 +1,424 @@
+                uniformMap : this._uniformMap,
+                boundingVolume : this._boundingSphere,
+                owner : this,
+                modelMatrix : Matrix4.IDENTITY,
+                pass : Pass.GROUND
+            });
+
+            var colorRenderState = context.createRenderState({
+                stencilTest : {
+                    enabled : true,
+                    frontFunction : StencilFunction.NOT_EQUAL,
+                    frontOperation : {
+                        fail : StencilOperation.KEEP,
+                        zFail : StencilOperation.KEEP,
+                        zPass : StencilOperation.DECREMENT
+                    },
+                    backFunction : StencilFunction.NOT_EQUAL,
+                    backOperation : {
+                        fail : StencilOperation.KEEP,
+                        zFail : StencilOperation.KEEP,
+                        zPass : StencilOperation.DECREMENT
+                    },
+                    reference : 0,
+                    mask : ~0
+                },
+                cull : {
+                    enabled : true,
+                    face : CullFace.BACK
+                },
+                depthTest : {
+                    enabled : true
+                },
+                depthMask : false,
+                blending : BlendingState.ALPHA_BLEND
+            });
+
+            this._colorPassCommand = new DrawCommand({
+                primitiveType : PrimitiveType.TRIANGLES,
+                vertexArray : this._va,
+                renderState : colorRenderState,
+                shaderProgram : this._sp,
+                uniformMap : this._uniformMap,
+                boundingVolume : this._boundingSphere,
+                owner : this,
+                modelMatrix : Matrix4.IDENTITY,
+                pass : Pass.GROUND
+            });
+        }
+
+        if (this.debugVolume && !defined(this._debugVolumeCommand)) {
+            var pauseRS = context.createRenderState({
+                cull : {
+                    enabled : false
+                },
+                depthTest : {
+                    enabled : true
+                },
+                depthMask : false,
+                blending : BlendingState.ALPHA_BLEND
+            });
+
+            var sp = context.createShaderProgram(ShadowVolumeVS, ShadowVolumeFS, attributeLocations);
+
+            this._debugVolumeCommand = new DrawCommand({
+                primitiveType : PrimitiveType.TRIANGLES,
+                vertexArray : this._va,
+                renderState : pauseRS,
+                shaderProgram : sp,
+                uniformMap : this._uniformMap,
+                boundingVolume : this._boundingSphere,
+                owner : this,
+                modelMatrix : Matrix4.IDENTITY,
+                pass : Pass.GROUND
+            });
+        }
+
+        var pass = frameState.passes;
+        if (pass.render) {
+            if (this.debugVolume) {
```

Comment:
Not sure how this will translate to `Primitive`, but I would call this `debugShowVolume`.